lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -1091,19 +1090,13 @@ static bool parseDeclSILOptional(bool *isTransparent,
 }
 
 bool SILParser::performTypeLocChecking(TypeLoc &T, bool IsSILType,
-                                       GenericEnvironment *GenericEnv,
-                                       DeclContext *DC) {
+                                       GenericEnvironment *GenericEnv) {
   if (GenericEnv == nullptr)
     GenericEnv = ContextGenericEnv;
 
-  if (!DC)
-    DC = &P.SF;
-  else if (!GenericEnv)
-    GenericEnv = DC->getGenericEnvironmentOfContext();
-
   return swift::performTypeLocChecking(P.Context, T,
                                        /*isSILMode=*/true, IsSILType,
-                                       GenericEnv, DC);
+                                       GenericEnv, &P.SF);
```

Comment:
This *may* weaken some of the checking that was there before, but I don't know enough about SIL offhand to be certain.

Reply:
Nope. This is only intended to find names at global scope. There was only one place where we passed in a DC other than the SourceFile -- we would pass in a ProtocolDecl. Not only was this unnecessary, since we weren't looking inside the protocols, it was precisely the thing that triggered your ASTScope assertion. :-)

Reply:
Thank you. That explains this change. I wonder why the ProtocolDecl was being passed in originally? Just a mistake?